lightning/src/ln/channelmanager.rs

```diff
@@ -3600,17 +3685,26 @@ where
 		let mut expected_amt_msat = None;
 		let mut valid_mpp = true;
 		let mut errs = Vec::new();
-		let mut channel_state = Some(self.channel_state.lock().unwrap());
+		let mut per_peer_state = Some(self.per_peer_state.read().unwrap());
```

Comment:
Ok sure :)!

Before changing this, I'd just like to I understand which behavior we'd like the keep:
1 . With the `read` lock only here, for non MPP payments, we can no longer be sure that the channel is still there when reaching the `claim_funds_from_hop` call. Is that something we'd like to avoid for non MPP payments?
2. For MPP payments, with the previous behavior we're sure that the the channel for the first source is still there when reaching the first `claim_funds_from_hop` call. I guess this shouldn't matter too much though right?

is it behavior (1.) we'd like to keep, to avoid `claim_funds_from_hop` failing for not finding the channel for non MPP payments :)?